lib/spack/spack/binary_distribution.py

```diff
@@ -269,7 +269,9 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
             raise NoOverwriteException(str(specfile_path))
     # make a copy of the install directory to work with
     workdir = os.path.join(tempfile.mkdtemp(), os.path.basename(spec.prefix))
-    install_tree(spec.prefix, workdir, symlinks=True)
+    # set symlinks=False here to avoid broken symlinks when archiving and
+    # moving the package
+    install_tree(spec.prefix, workdir, symlinks=False)
```

Comment:
In the context of https://github.com/spack/spack/issues/10062#issuecomment-445453848, why `symlinks=False` is required.

Reply:
Some of the links in the bzip2 package used absolute paths. When deploying the buildcache, link creation would fail because the target paths were missing. This was a temporary solution. The correct solution would be to make the links relative when creating the buildcache.

Reply:
I don't think above is solving bzip2 issue with buildcache. I had trouble yesterday while testing buildcache. (bzip2 binary linked to libbz2.so is installed without rpath)
  
> The correct solution would be to make the links relative when creating the buildcache

agree.